samples/nrf9160/modem_shell/src/link/link_shell.c

```diff
@@ -1280,9 +1301,39 @@ int link_shell(const struct shell *shell, size_t argc, char **argv)
 			link_modem_tau_notifications_unsubscribe();
 		}
 		break;
-	case LINK_CMD_CONNECT:
-		ret = link_shell_pdn_connect(shell, apn, family);
-		break;
+	case LINK_CMD_CONNECT: {
+		struct link_shell_pdn_auth auth_params;
```

Comment:
This is a bit of a common problem in this file that there are quite long code blocks within this switch case. I'd like to have this type of code handled in separate functions.

Reply:
Yes, agree, a common problem in this file. Parsing is done mainly here, and not sure does splitting in this case makes this better. Left as it is for now.